bitswap/client/internal/messagequeue/donthavetimeoutmgr_test.go

```diff
@@ -72,6 +72,26 @@ func (tr *timeoutRecorder) clear() {
 	tr.timedOutKs = nil
 }
 
+// quartzAdd advances the time by the given duration through any events that
+// happen before or at the final time.
+//
+// TODO: Replace calls to quartzAdd with calls to clock.Advance and
+// clock.AdvanceNext to provide better control over time events.
+func quartzAdd(c *quartz.Mock, d time.Duration) {
+	for d > 0 {
+		next, ok := c.Peek()
+		if !ok || next >= d {
+			c.Advance(d)
+			time.Sleep(time.Millisecond)
```

Comment:
```suggestion
			c.Advance(d).MustWait(context.Background())
```

`MustWait()` returns once all triggered events are complete, which is faster to run and more accurate than sleeping.

Reply:
I needed to use `time.Sleep` here. When I use `MustWait` that causes tests to occasionally hang forever. I _think_ that `MustWati` is waiting for completion of an `AfterFunc` that itself is waiting for another timer. Using the sleep here avoids that situation.

Please feel free to investigate this, because using `MustWait` would certainly be preferable.